### PR TITLE
CentOS 8 Support

### PR DESCRIPTION
**Pull request by jcsiadal**
CentOS 8 support added to Warewulf

Commits from original PR #203 squashed into this.

This commit does the following:

o   Adds a build for CentOS 8 to CircleCI.
    Cross-compile options are not available in epel8,
    so this options is disabled for now. The RPM specfiles were modified to
    support configure options needed by CentOS8 and SLE15.

o   The Warewulf version was fixed to 3.9.0.

o   Specfiles *.in files updated to support CircleCI changes

o   ntpd init updated. The ntp package is not supported in CentOS 8.
    The init is updated so if ntp is not installed, wwinit doesn't create
    a config by default. A chrony init will need to be created, which will
    run first and skip configuration if ntp is installed.
    OpenNTP is also now supported.

o   Added a template for CentOS 8 nodes and updated the includes template
    - Added DNF support. A flag is used in the template; otherwise it defaults to YUM
    - A dracut workaround was needed to fix copying the kernel to boot. The new
      RHEL8 now uses /sys and /proc to set kernel configuration, which is
      useless for hosting different node image architectures.

o   Added check for sysmacros.ph used by wwvnfs It no longer exists in glibc 2.26.

o    includes copy of my CentOS8 build script in platform. For reference only.

One big change here was I decoupled the chroot package manager configuration from the host pkgmgr configuration. WW 3.8 copies the head node YUM config/repos to the chroot before installation, using a temporary YUM config file. I removed that. Instead, it makes the repos and configuration used during chroot installation permanent within that chroot. The head node and compute node repos remain separate. This allows users on a CentOS 8 or RHEL 8 head node to easily build CentOS 6, 7, or 8 chroot images. I tested building all three of these.

### CODE REVIEW

File: ipmi/warewulf-ipmi.spec.in [hunk 1]
```diff
@@ -13,8 +13,16 @@ URL: http://warewulf.lbl.gov/
...
+%if 0%{?rhel} >= 8 || 0%{?sle_version} >= 150000
+BuildRequires: ipmitool
+Requires: ipmitool
+%define CONF_FLAGS "--with-local-ipmitool=yes"
+%endif
+
 BuildRequires: warewulf-common openssl-devel
-Conflicts: warewulf < 3
+BuildRequires: autoconf, automake
+Conflicts: warewulf < 3.9
```
Comment (bensallen) [inline ipmi/warewulf-ipmi.spec.in]: Curious why this conflicts with < 3.9?

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,83 +1,197 @@
...
-
-    working_directory: ~/warewulf3
-
+      -
+        image: "centos:7"
     steps:
       - checkout
-      - run:
-          name: Install Build Dependencies
+      -
```
Comment (bensallen) [inline .circleci/config.yml:8]: This is just a nit, but there's a handful of instances where there's a list `- ` with a newline. Can the extra newline be removed. In this instance for example, `run:` should be `- run:`.

File: ipmi/warewulf-ipmi.spec.in (hunk 1 above)
Comment (jcsiadal) [reply ipmi/warewulf-ipmi.spec.in]: I started adjusting during my initial scrub, thinking there was a virtual warewulf package in one of the specs. After I realized warewulf 2 was issued as a package, it made sense and I thought I reverted these to 3. In any case, I'll issue a quick commit to fix.

File: .circleci/config.yml (hunk 2 above)
Comment (jcsiadal) [reply .circleci/config.yml:8]: I copy/pasted the yamllint output. I prefer it your way too. The linter rule is that a key/value in a sequence is on the same line, with the exception that a block mapping in a sequence gets a newline after the hyphen.
Let me know if we need to fix it, although I do like the ability for auto-formatters do the work for me.

Comment (bensallen) [reply .circleci/config.yml:8]: If yamllint likes it I guess its good enough for me too. Thanks.

**Comment by bensallen** [comment]
Looks good to me. @jmstover any issues with this?

Ben

**Review by bensallen — APPROVED** [review]

**Comment by bensallen** [comment]
Thanks @jcsiadal for all the hard work on this on!

**Pull request merged**